backend/src/crawler/udn_crawler.py

```diff
@@ -0,0 +1,148 @@
+        crawler.save(news, db_session)
+
+UDNCrawler Methods:
+    __init__(self, timeout: int = 5): Initializes the crawler with a default timeout for HTTP requests.
+    startup(self, search_term: str) -> list[Headline]: Fetches news headlines for a given search term across multiple pages.
+    get_headline(self, search_term: str, page: int | tuple[int, int]) -> list[Headline]: Fetches news headlines for specified pages.
+    _fetch_news(self, page: int, search_term: str) -> list[Headline]: Helper method to fetch news headlines for a specific page.
+    _create_search_params(self, page: int, search_term: str): Creates the parameters for the search request.
+    _perform_request(self, params: dict): Performs the HTTP request to fetch news data.
+    _parse_headlines(response): Parses the response to extract headlines.
+    parse(self, url: str) -> News: Parses a news article from a given URL.
+    _extract_news(soup, url: str) -> News: Extracts news details from the BeautifulSoup object.
+    save(self, news: News, db: Session): Saves a news article to the database.
+    _commit_changes(db: Session): Commits the changes to the database with error handling.
+"""
+
+from requests import Response
+from bs4 import BeautifulSoup
+from sqlalchemy.orm import Session
+import requests
+
+from .crawler_base import NewsCrawlerBase, Headline, News, NewsWithSummary
+
+
+class UDNCrawler(NewsCrawlerBase):
+    CHANNEL_ID = 2
+
+    def __init__(self, timeout: int = 5) -> None:
+        self.news_website_url = "https://udn.com/api/more"
+        self.timeout = timeout
+
+    def startup(self, search_term: str) -> list[Headline]:
+        """
+        Initializes the application by fetching news headlines for a given search term across multiple pages.
+        This method is typically called at the beginning of the program when there is no data available,
+        hence it fetches headlines from the first 10 pages.
+
+        :param search_term: The term to search for in news headlines.
+        :return: A list of Headline namedtuples containing the title and URL of news articles.
+        :rtype: list[Headline]
+        """
+        return self.get_headline(search_term, page=(1, 10))
+
+    def get_headline(
+        self, search_term: str, page: int | tuple[int, int]
+    ) -> list[Headline]:
+
+        # Calculate the range of pages to fetch news from.
+        # If 'page' is a tuple, unpack it and create a range representing those pages (inclusive).
+        # If 'page' is an int, create a list containing only that single page number.
+        # page_range = range(*page) if isinstance(page, tuple) else [page]
+        page_range = range(*page) if isinstance(page, tuple) else [page]
+
+        headlines = []
+
+        for page_num in page_range:
+            news = self._fetch_news(page_num, search_term)
+            headlines.extend(news)
+
+    def _fetch_news(self, page: int, search_term: str) -> list[Headline]:
```

Comment:
This method should return a list of `Headline` rather than a list of `News`, I'd suggestion you rename it to `_fetch_headlines`.
```suggestion
    def _fetch_headlines(self, page: int, search_term: str) -> list[Headline]:
```